cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -663,6 +669,78 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
     return cw.toByteArray();
   }
 
+  /**
+   * Rewrites the temporary file name used for checkpoints from checkpointDir/temp to
+   * checkpointDir/temp-(current_timestamp_milliseconds).
+   *
+   * @param byteCodeStream {@link InputStream} for reading in the original bytecode.
+   * @return the rewritten bytecode
+   */
+  private byte[] rewriteTempFileNameForCheckpoint(InputStream byteCodeStream) throws IOException {
+    final Type hadoopPathType = Type.getType(org.apache.hadoop.fs.Path.class);
+    final Type stringType = Type.getType(java.lang.String.class);
+    final Type stringBuilderType = Type.getType(java.lang.StringBuilder.class);
+    ClassReader cr = new ClassReader(byteCodeStream);
+    ClassWriter cw = new ClassWriter(0);
+
+    cr.accept(new ClassVisitor(Opcodes.ASM5, cw) {
+      @Override
+      public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+
+        MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
+
+        return new MethodVisitor(Opcodes.ASM5, mv) {
+
+          @Override
+          public void visitMethodInsn(int opcode, String owner, String name, String desc, boolean itf) {
+
+            // If we see a call to the Path constructor with 2 String parameters, rewrite the second argument
+            // to add a timestamp.
+            if (opcode == Opcodes.INVOKESPECIAL &&
+              name.equals("<init>") &&
```

Comment:
Should compare with the method signature.

```java
Method method = new Method("<init>", Type.VOID_TYPE, new Type[]{ stringType, StringType });
...
if (method.equals(new Method(name, desc))) {
   // Matched
}
```

Reply:
Also, I would suggest check if the previous instruction was a `LDC "temp"`. We don't want to rewrite if the second parameter is not the constant string `"temp"`

Reply:
@chtyim I have added logic to check wether the previous instruction to the Path constructor call is the `LDC "temp"` instruction.